docs/designs/scrapping-workflow-phase-1-design.md

```diff
@@ -0,0 +1,67 @@
+At first, if the input is CSV, this can be a simple csv uploader script.
+
+2. Angostura does it’s processing and persists the data to a big query table
+
+3. A task defined in airflow queries for yet-to-be-enriched data and proceeds to scrape.
+It can then persists the output to the table that classification workflow(s) will start from. 
+   
+    More on this job below. 
+
+* Any further tasks such as cleaning and classification can be done further along the pipeline.
+
+**Scraping task** 
+
+* Here, we can leverage the python_transform job type Angostura offers. (which is backed by the BigQueryPythonOperator). From the Angostura documentation:
+> The python_transform job takes a list of SQL files as arguments, a python function as a string and writes the result of calling that function with the arguments into a target BIgQuery table [...] Each transformation function is a Python function that receives pandas data frames as arguments.
+
+* This means we could write a python module which     
+    * Exposes one function, which takes a data frame as input and produces a data frame as output.
+    * The input consists of the rows of data that need scraping. 
+    * The output can be the same rows, with the scraped data now present in the columns where it should be so that they can be persisted.
+    
+_Then the job becomes_
+1. Query for all the rows of data that haven’t been scrapped yet. 
+   * As a simple example, if there is a nullable field called scraped_data in the persisted data, then the query can simply be select * where scraped_data is null
+   * The query can also be modified to select only rows that contain a specific source, and pass those to a specific scraping task etc. That should make it flexible enough for anything we want to do in phase 2.
+
+2. The python function exposed by the module is called with these rows passed in.
+3. Scraping happens.
+4. Scraping output is persisted.
+ 
+**The python module** 
+
+This can be very simple for the MVP. 
+A sample design can simply be to:
+* Define a scraper interface, which takes in a URL (+ anything else needed) and outputs scraped data.
+* Implement a top level method, which contains a map of URL domains to scraper implementations
+* Output is written to data frame
+
+This is modularised, and lets us write specific scrapers for specific sources easily and independently of each other. We can only start with a few scrapers for specific, high-priority data sources.
+
+**Picking the first source to write a scraper for**
+
+From the excel [_TODO: add link_] that was provided by the partner NGO, we've worked out that the following distribution of sources:
+
+![top_20_sources](../images/top_20_sources_osvp_excel.png)
+
+This suggests that we should start writing a scraper for [El Pitazo](https://elpitazo.net/) first. 
+
+Of course, we should confirm with the _servicios publicos_ team what they and the ngo see as the highest priority sources. 
```

Comment:
@marianelamin we could probably start engaging the team with regards to the sources they'd like us to consider as a priority - what do you think?

Reply:
Yeah. Not sure if we have formally introduced the project to the servicio público leads.
👀 @Edilmo @dieko95

Reply:
@marianelamin I think it would be great to start this conversation with the ngo + _services publicos_ to be 100% sure that all of us are on the same page.

Reply:
let's do it 🥳

Reply:
Can I resolve this? Is the task registered, well, somewhere?

Reply:
We can resolve this.
I will coordinate to have a formal introduction of the project with our vertical first: _Servicios Publicos_ ❤️

Reply:
perfect @marianelamin